OpenRA.Mods.D2k/Traits/Sandworm.cs

```diff
@@ -99,11 +100,10 @@ void RescanForTargets(Actor self)
 				if (!actor.IsInWorld)
 					continue;
 
-				// TODO: Test if we really want to ignore actors that are on rock
 				if (!mobile.Value.CanEnterCell(actor.Location, null, false))
 					continue;
 
-				var noise = actor.TraitOrDefault<AttractsWorms>();
+				var noise = actor.TraitsImplementing<AttractsWorms>().FirstOrDefault(x => !x.IsTraitDisabled);
```

Comment:
Could we aggregate the noise over all the enabled traits instead?  That would bring this more in line with things like power and *Modifiers.